### PR TITLE
autolock expired accounts

### PR DESCRIPTION
**Pull request by jacobkagon**
[ME-186972351](https://www.pivotaltracker.com/n/projects/2640060/stories/186972351)

**Please look at devise.rb and expired_login_attempt_controller_spec.rb**

## Summary by CodeRabbit

- **New Features**
	- Integrated `devise-security` for enhanced user account management features.
	- Implemented user authentication handling across the application.
	- Added new functionalities for user tracking and session management.
	- Enhanced flexibility in configuring account lock periods for users.
- **Documentation**
	- Added comprehensive descriptions and responsibilities for various publisher classes and operations.
	- Enhanced attribute documentation for several models.
- **Refactor**
	- Simplified control flow in operations by removing explicit `return` statements.
	- Updated namespace references and simplified file reading processes in specs and scripts.
- **Chores**
	- Updated the logic for calculating transactions per iteration in IRS 1095A CSV report generation.
- **Tests**
	- Introduced controller spec for handling login attempts of expired users, including behavior tests.

### CODE REVIEW

File: app/operations/fdsh/esi/rj14/request_json_esi_determination.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@
           event = PublishEventStruct.new(PUBLISH_EVENT, params[:transmittable_objects][:transaction].json_payload,
                                          { authorization: "Bearer #{params[:token]}",
                                            messageid: params[:transmittable_objects][:job].message_id,
-                                           partnerid: ENV['CMS_PARTNER_ID'] })
+                                           partnerid: ENV.fetch('CMS_PARTNER_ID', nil) })
```
[the remark this answers is not in the document]
Comment (saikumar9) [reply app/operations/fdsh/esi/rj14/request_json_esi_determination.rb:45]: > Use of `ENV.fetch` with a default value of `nil` for `partnerid` is redundant. Simplify by directly accessing the environment variable.
> 
> ```diff
> -                                           partnerid: ENV.fetch('CMS_PARTNER_ID', nil) })
> +                                           partnerid: ENV['CMS_PARTNER_ID'] })
> ```
> 
> Committable suggestion

@coderabbitai How can I make you not raise/comment when I use `ENV.fetch` and pass `nil` as a default value?

File: app/operations/fdsh/esi/h14/request_esi_determination.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ class RequestEsiDetermination
         include Dry::Monads[:result, :do, :try]
         include EventSource::Command
 
-        PublishEventStruct = Struct.new(:name, :payload, :headers)
+        PublishEventStruct = Struct.new(:name, :payload, :headers, :message)
```
[the remark this answers is not in the document]
Comment (saikumar9) [reply app/operations/fdsh/esi/h14/request_esi_determination.rb:11]: This field is being used in the `event_source` gem for integration.

File: app/operations/fdsh/non_esi/rj31/request_json_non_esi_determination.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ class RequestJsonNonEsiDetermination
         include Dry::Monads[:result, :do, :try]
         include EventSource::Command
 
-        PublishEventStruct = Struct.new(:name, :payload, :headers)
+        PublishEventStruct = Struct.new(:name, :payload, :headers, :message)
```
[the remark this answers is not in the document]
Comment (saikumar9) [reply app/operations/fdsh/non_esi/rj31/request_json_non_esi_determination.rb:11]: The `message` field is used internally in the `event_source` gem for publishing the `payload`.

File: app/event_source/publishers/fdsh/eligibilities/hub_connectivity_publisher.rb [hunk 4]
```diff
@@ -3,6 +3,9 @@
 module Publishers
   module Fdsh
     module Eligibilities
+      # The HubConnectivityPublisher class is responsible for publishing events related to Hub connectivity.
+      # It includes the EventSource::Publisher module to enable event publishing.
+      # It registers an event: 'hub_connectivity_check_complete'.
```
[the remark this answers is not in the document]
Comment (saikumar9) [reply app/event_source/publishers/fdsh/eligibilities/hub_connectivity_publisher.rb]: Updated

**Review by vkghub — APPROVED** [review]
LGTM!

**Review by jayreddy519 — APPROVED** [review]

**Pull request merged**